From 9e5d3e169c5db007caf5fd3430645ae94a26bac8 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 24 May 2024 12:37:43 +1000
Subject: [PATCH] include: Complete __wine_uuidof for C++
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/usr/local/include/wine/windows/guiddef.h:83:68: error: incomplete type ‘__wine_uuidof_type<IUnknown>::inst’ {aka ‘__wine_uuidof<IUnknown>’} used in nested name specifier
   83 | #define __uuidof(type) __wine_uuidof_type<__typeof__(type)>::inst::uuid
---
 include/guiddef.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/guiddef.h b/include/guiddef.h
index c5ba9635b89..15c6b13b55d 100644
--- a/include/guiddef.h
+++ b/include/guiddef.h
@@ -56,7 +56,9 @@ typedef struct _GUID
 #ifdef __WINE_UUID_ATTR
 
 extern "C++" {
-    template<typename T> struct __wine_uuidof;
+    template<typename T> struct __wine_uuidof { \
+            static const GUID uuid;                                     \
+        };
 
     template<typename T> struct __wine_uuidof_type {
         typedef __wine_uuidof<T> inst;
-- 
2.43.0

